backend/src/openbeheer/api/views.py

```diff
@@ -87,7 +112,92 @@ def get_renderers(self) -> list[BaseRenderer]:
         return [MsgspecJSONRenderer()] + [r() for r in render_classes]
 
 
-class ListView[P: OBPagedQueryParams, T](MsgspecAPIView):
+def mk_expandable[T: Type[Struct]](t: T, fields: Mapping[str, type]) -> T:
```

Comment:
I think that we have different preferences for naming things, I think I would go for `make_expanded_type` here :grin:

Reply:
What about `make_expandable_type`, as it's not expanded yet.